app/models/metric/capture.rb

```diff
@@ -31,210 +31,49 @@ def self.concurrent_requests(interval_name)
-      interval = method_name.sub("perf_capture_", "")
-      msg = "#{items.length} #{interval.inspect} captures on the queue for zone [#{zone.name}]"
-      msg << " - oldest: [#{items.first.created_on.utc.iso8601}], recent: [#{items.last.created_on.utc.iso8601}]" if items.length > 0
-      _log.info(msg)
-    end
-  end
-  private_class_method :perf_capture_health_check
-
-  # Collect realtime targets and group them by their rollup parent
-  #
-  # 1. Only calculate rollups for Hosts
-  # 2. Some Hosts have an EmsCluster as a parent, others have none.
-  # 3. Only Hosts with a parent are rolled up.
-  # @param [Array<Host|VmOrTemplate|Storage>] @targets The nodes to rollup
-  # @option options :force Force capture if this node is a host
-  # @returns Hash<String,Array<Host>>
-  #   e.g.: {"EmsCluster:4"=>[Host:4], "EmsCluster:5"=>[Host:1, Host:2]}
-  def self.calc_targets_by_rollup_parent(targets, options = {})
-    realtime_targets = targets.select do |target|
-      target.kind_of?(Host) &&
-        perf_target_to_interval_name(target) == "realtime" &&
-        (options[:force] || perf_capture_now?(target))
-    end
-    realtime_targets.each_with_object({}) do |target, h|
-      target.perf_rollup_parents("realtime").to_a.compact.each do |parent|
-        pkey = "#{parent.class}:#{parent.id}"
-        (h[pkey] ||= []) << target
-      end
-    end
-  end
-
-  # Determine queue options for each target
-  # Is only generating options for Vmware Hosts, which have a task for rollups.
-  # The rest just set the zone
-  def self.calc_target_options(zone, targets_by_rollup_parent)
-    task_end_time           = Time.now.utc.iso8601
-    default_task_start_time = 1.hour.ago.utc.iso8601
-
-    target_options = Hash.new { |h, k| h[k] = {:zone => zone} }
-    # Create a new task for each rollup parent
-    # mark each target with the rollup parent
-    targets_by_rollup_parent.each_with_object(target_options) do |(pkey, targets), h|
-      name = "Performance rollup for #{pkey}"
-      prev_task = MiqTask.where(:identifier => pkey).order("id DESC").first
-      task_start_time = prev_task ? prev_task.context_data[:end] : default_task_start_time
-
-      task = MiqTask.create(
-        :name         => name,
-        :identifier   => pkey,
-        :state        => MiqTask::STATE_QUEUED,
-        :status       => MiqTask::STATUS_OK,
-        :message      => "Task has been queued",
-        :context_data => {
-          :start    => task_start_time,
-          :end      => task_end_time,
-          :parent   => pkey,
-          :targets  => targets.map { |target| "#{target.class}:#{target.id}" },
-          :complete => [],
-          :interval => "realtime"
-        }
+    zone ||= MiqServer.my_server(true).zone
+    perf_collect_all_metrics_queue(zone.ext_management_systems, "realtime")
+  end
+
+  def self.perf_collect_all_metrics_queue(ems, interval_name = "realtime", start_time = nil, end_time = nil, options = {})
```

Comment:
question: why not make the param `ems` as `emses` and we have no need for the `Array.wrap`?
(Is this ruby style to make the caller happier?)

Reply:
I had not realized that everything coming in is an array.

My initial intent was for us to queue a single metrics collect.
Sad to see the `Array.wrap` go, but think you are right